pkg/controller/sparkapplication/submission.go

```diff
@@ -94,6 +94,9 @@ func buildSubmissionCommandArgs(app *v1beta2.SparkApplication, driverPodName str
 		return nil, err
 	}
 
+	// Add proxy user
+	args = append(args, addProxyUserOptions(app)...)
```

Comment:
Minor, maybe still try to keep --mater at the top? overall looking good.

Reply:
Done.